target/linux/ramips/dts/mt7621_winstars_ws-wn583a6.dts

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "winstars,ws-wn583a6", "mediatek,mt7621-soc";
+	model = "Winstars WS-WN583A6";
+
+	aliases {
+		led-boot = &led_status_red;
+		led-failsafe = &led_status_red;
+		led-running = &led_status_blue;
+		led-upgrade = &led_status_red;
+		label-mac-device = &wifi2g;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,57600";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		night_light_white {
+			label = "ws-wn583a6:white:night_light";
+			gpios = <&gpio 15 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_status_blue: status_blue {
+			label = "ws-wn583a6:blue:status";
+			gpios = <&gpio 22 GPIO_ACTIVE_LOW>;
+		};
+
+		status_amber {
+			label = "ws-wn583a6:amber:status";
+			gpios = <&gpio 24 GPIO_ACTIVE_LOW>;
+		};
+
+		led_status_red: status_red {
+			label = "ws-wn583a6:red:status";
+			gpios = <&gpio 25 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <104000000>;
```

Comment:
And you are sure that this is stable?

Reply:
I flashed it many times without problems. I also did some tests without any error in the dmesg:
```
root@OpenWrt:~# time dd if=/dev/mtd3 of=/dev/null
15744+0 records in
15744+0 records out
real	0m 1.86s
user	0m 0.03s
sys	0m 1.82s
root@OpenWrt:~# time dd if=/dev/zero of=/root/output.file bs=1024 count=1024
1024+0 records in
1024+0 records out
real	0m 10.06s
user	0m 0.00s
sys	0m 10.02s
```
Looking into the datasheet this speed should be safe:
![image](https://user-images.githubusercontent.com/2470733/88457929-5845ce80-ce8a-11ea-827b-66d2a99ca1e2.png)
However if you think that we should play safe, I will lower the speed :wink: